src/compiler.ts

```diff
@@ -3135,7 +3141,7 @@ export class Compiler extends DiagnosticEmitter {
         }
         if (initExpr) {
           initializers.push(
-            this.makeLocalAssignment(local, initExpr, type, false)
+            this.makeLocalAssignment(local, initExpr, initType ? initType : type, false)
```

Comment:
Doesn't this work `initType || type` ?

Reply:
It works. Let me change all the similar situation by this clear and simple way.

Reply:
In both main and feature branch, it does not work.
It needs to do some special process for 'or' operation